service/matching/tasklist/matcher.go

```diff
@@ -480,7 +482,9 @@ func (tm *taskMatcherImpl) pollOrForward(
 		if task.ResponseC != nil {
 			tm.scope.IncCounter(metrics.PollSuccessWithSyncPerTaskListCounter)
 		}
-		tm.scope.RecordTimer(metrics.PollLocalMatchLatencyPerTaskList, time.Since(startT))
```

Comment:
It's less error-prone to add poll wait latency in the main Poll function. 

Line 432
```
task, err := tm.pollOrForward(ctxWithCancelPropagation, startT, isolationGroup, isolatedTaskC, tm.taskC, tm.queryTaskC)
if task != nil {
    task.PollLatency = time.since(startTime)
}
```